File: solutions/javascript/perfect-numbers/1/perfect-numbers.js
//
// This is only a SKELETON file for the 'Perfect Numbers' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const classify = (x) => {
  if (x < 1)
    throw new Error("Classification is only possible for natural numbers.");
  if (x === 1) return "deficient";
  let y = x;
  let numbers = [1];
  for (let n = 2; n < x; n++) {
    if (x % n === 0) {
      numbers.push(n);
    }
  }

  y = numbers.reduce((acc, cur) => acc + cur);
  return y === x ? "perfect" : y < x ? "deficient" : "abundant";
};

File: solutions/python/perfect-numbers/1/perfect_numbers.py
def classify(n):
    """ A perfect number equals the sum of its positive divisors.

    :param number: int a positive integer
    :return: str the classification of the input integer
    """
    if n < 1:
        raise ValueError("Classification is only possible for positive integers.")
        
    suma_alicuota = sum([i for i in range(1, n) if n % i == 0])
    
    if suma_alicuota == n:
        return "perfect"
    elif suma_alicuota > n:
        return "abundant"
    else:
        return "deficient"